app/lib/widgets/app_bar/app_bar.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flex_ui/flex_ui.dart';
+import 'package:flex_ui/tokens/sizes.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_annotation/widgetbook_annotation.dart' as widgetbook;
+
+class FlexAppBar extends StatelessWidget implements PreferredSizeWidget {
+  FlexAppBar(
+      {super.key,
+      this.leadingIcon,
+      this.onLeadingIconPressed,
+      this.title,
+      this.trailingIcon,
+      this.onTrailingIconPressed,
+      this.centerTitle = true})
+      : assert(
+          (leadingIcon == null) == (onLeadingIconPressed == null),
+          'If leadingIcon is not null, onLeadingIconPressed must not be null, and vice versa.',
+        ),
+        assert(
+          (trailingIcon == null) == (onTrailingIconPressed == null),
+          'If trailingIcon is not null, onTrailingIconPressed must not be null, and vice versa.',
+        );
+
+  final Widget? title;
+  final IconData? leadingIcon;
+  final VoidCallback? onLeadingIconPressed;
+  final IconData? trailingIcon;
+  final VoidCallback? onTrailingIconPressed;
+  final bool? centerTitle;
+
+  @override
+  Widget build(BuildContext context) {
+    final brightness = Theme.of(context).brightness;
+    final iconButtonStyle = Theme.of(context).iconButtonTheme.style?.copyWith(
+        backgroundColor: WidgetStatePropertyAll(Colors.transparent),
+        iconColor: WidgetStatePropertyAll(
+            brightness == Brightness.dark ? Colors.white : Colors.black),
```

Comment:
overriding default theme is fine but let's discuss how we want to handle injecting `FlexColors` into these widgets.. maybe we use a hybrid approach, for example:

```
brightness == Brightness.dark 
        ? Theme.of(context).colorScheme.onSurface 
        : Theme.of(context).colorScheme.onBackground
```

Reply:
Yes I'm agree we need to improve the default Theme to be aligned. For this feature, I've kept it simple as I can